packages/headless/src/api/search/html/html-api-client.ts

```diff
@@ -27,6 +27,29 @@ export interface HtmlAPIClientOptions {
   requestMetadata?: RequestMetadata;
 }
 
+export const buildSrcPath = (req: HtmlRequest, path: string) => {
+  const url = new URL(`${req.url}${path}`);
+  url.searchParams.append('access_token', req.accessToken);
+  url.searchParams.append('organizationId', req.organizationId);
+  url.searchParams.append('uniqueId', req.uniqueId);
+  if (req.q !== undefined) {
+    url.searchParams.append('q', req.q);
+  }
+  if (req.enableNavigation !== undefined) {
+    url.searchParams.append('enableNavigation', `${req.enableNavigation}`);
+  }
+  if (req.requestedOutputSize !== undefined) {
+    url.searchParams.append(
+      'requestedOutputSize',
+      `${req.requestedOutputSize}`
+    );
+  }
+  if (req.visitorId !== undefined) {
+    url.searchParams.append('visitorId', `${req.visitorId}`);
+  }
+  return url.href;
+};
```

Comment:
I'm not sure if we should use the `URL` object, since we aim to make Headless as platform-agnostic as possible and it looks like React Native has [some differences](https://www.npmjs.com/package/react-native-url-polyfill#user-content-why-do-we-need-this) with its `URL` implementation.

Reply:
Hmmm would you prefer straight up string interpolation? That felt sketchy.

Reply:
I added a utility class to avoid using URL. Let me know what you think.